src/taxpasta/infrastructure/application/diamond/diamond_profile.py

```diff
@@ -0,0 +1,35 @@
+# Copyright (c) 2022, Moritz E. Beber, Maxime Borry, Jianhong Ou, Sofia Stamouli.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+"""Provide a description of the diamond profile format."""
+
+
+import pandera as pa
+from pandera.typing import Series
+
+
+class DiamondProfile(pa.SchemaModel):
+    """Define the expected diamond profile format."""
+
+    query_id: Series[str] = pa.Field()
+    taxonomy_id: Series[int] = pa.Field(ge=0)
```

Comment:
I think it's better to keep taxonomic identifiers as `str` rather than integers.

Reply:
Hmm, I first changed it to string, but I'm not sure this is relevant, because even GTDB based taxonomy need to be provided with `names.dmp` and `nodes.dmp` file which use integers TAXIDs (see https://github.com/shenwei356/gtdb-taxdump for example).
This also ensures that TAXIDs are not set to negative integers.

Reply:
I think both is reasonable. My motivation to suggest strings or categoricals is that you would never want to do arithmetic with taxonomy identifiers. So parsing them as string ensures that this is explicit. For the standard profile or output from taxpasta we might also want to use identifiers with prefixes like `NCBI:txid4607`. But if the DIAMOND input requires numeric tax IDs then it's also reasonable to parse them that way and ensure that they're not negative. 👍🏼 Just make sure to convert them appropriately for the standard profile. And we need to start thinking about reading in a matching taxonomy into taxpasta (we should actually ensure that all data read in is contained in the given taxonomy) and solve unclassified reads within those taxonomies.

Reply:
I've casted them back to `str` in the profile standardisation